File: 2022/day_02/haskell/app/Main.hs
module Main (main) where

import Game ( getRoundPoints, lineToRound )

main :: IO()
main = do
    inputFile <- readFile "../input.txt"
    let run problemPart = sum $ map (getRoundPoints . lineToRound problemPart) (lines inputFile)
    let result1 = run 1
    let result2 = run 2
    print ("Part 1 result: " ++ show result1)
    print ("Part 1 result: " ++ show result2)

File: 2022/day_02/rust/src/main.rs
use game::{lines_to_rounds, sum_points_for_rounds};
use std::fs;

mod game;

/// Reads problem input from file.
fn read_input_file() -> Vec<String> {
    let input_file_path = "../input.txt";
    let mut input_file_lines: Vec<String> = Vec::new();
    for line in fs::read_to_string(input_file_path).unwrap().lines() {
        input_file_lines.push(line.to_string());
    }
    input_file_lines
}

fn main() {
    let input_file_lines = read_input_file();
    let rounds_part_1 = lines_to_rounds(&input_file_lines, 1);
    let result_1 = sum_points_for_rounds(rounds_part_1);
    let rounds_part_2 = lines_to_rounds(&input_file_lines, 2);
    let result_2 = sum_points_for_rounds(rounds_part_2);
    println!("Part 1 answer: {}", result_1);
    println!("Part 2 answer: {}", result_2);
}
